### DISCUSSION TITLE
How to display URL image on localhost editor

### DISCUSSION THREAD

**Discussion by ahmedqfarrukh** [gollum/gollum#1977] [Q&A] [answered]
Hey all I cannot figure out how to properly display an image from URL with gollum locally. The images urls work on incognito browser  so no permissioning needed. Just using `gollum` to start and stop local server in the cloned wiki directory. Have tried enabling `--allow-uploads` `page` and `dir` and it didn't work. The image files are also stored in the `[wiki_directory]/images` folder locally. This is what it appears like:
<img width="787" alt="Screen Shot 2023-06-12 at 9 59 18 AM" src="https://github.com/gollum/gollum/assets/133700651/34cd5bdd-f557-4d65-b070-a841fac5f12e">

Markdown:
```
# Red Mountain Makers Wiki
<img src="https://raw.githubusercontent.com/wiki/redmountainmakers/wiki/images/rmm_logo_2023.png" width=15% height=15%>

## Welcome to the Red Mountain Makers Wiki
Our mission as a technical cooperative is to engage in scientific and artistic research, experimentation, and education. To do so, we are adapting a workspace for a collaborative and supportive community of peer and project based education related to technology and artistic expression. We also support the use and development of free and opensource software, hardware and design.


<img src="https://raw.githubusercontent.com/wiki/redmountainmakers/wiki/images/directions-to-rmm.png" width=50% height=50%>

---
<img src="https://raw.githubusercontent.com/wiki/redmountainmakers/wiki/images/rmm_layout.png" width=50% height=50%>
```

Machine:
MacBook Air (M1, 2020)
macOS Monterey 12.1
ruby 2.7.3p183
Gollum 5.3.2

Any and all help is much appreciated!

**Comment by benjaminwil** [comment, marked as answer]
Hey there,

I was able to get the images to load by adding quotations marks around the height and width attributes:

```
<img src="https://raw.githubusercontent.com/wiki/redmountainmakers/wiki/images/rmm_layout.png" width="50%" height="50%">
```

Though not absolutely necessary, you may want to make your `<img>` tags self-closing as well by adding a slash to the end of the tag:

```
<img src="https://raw.githubusercontent.com/wiki/redmountainmakers/wiki/images/rmm_layout.png" width="50%" height="50%" />
```

Hope that helps!

**Reply by ahmedqfarrukh** [reply]
Thank you for your time and help. This also helped me understand that gollum is git-commit based and it is working on my machine as well. Do you have any advice on viewing markdown changes when you edit them on a local git-cloned wiki that wouldn't require commit for every change?

**Reply by benjaminwil** [reply]
>  Do you have any advice on viewing markdown changes when you edit them on a local git-cloned wiki that wouldn't require commit for every change?

The preview tab on the editor should render any Markdown changes properly without committing.

<img width="520" alt="Screenshot 2023-06-12 at 11 53 25 AM" src="https://github.com/gollum/gollum/assets/883581/5ede531a-de79-4216-a1e2-0f3e5d94671f">

**Reply by ahmedqfarrukh** [reply]
I see; however, I am mainly creating changes in the local markdown files in VS Code. Unsure if this is proper use of gollum, but ultimately I was hoping to develop the wiki by saving markdown files and then refreshing/recreating localhost and it being reflected in the browser. Would you have any insight on accomplishing this or any other streamlined editing?

**Reply by benjaminwil** [reply]
I see what you mean now, thanks for explaining.

I don't think that your editing workflow is aligned with what Gollum was developed for. But since Markdown and HTML are relatively universal, I think you can still preview your rendered Markdown before generating commits in Gollum.

For example, there are many VSCode plugins that would let you accurately preview and validate any Markdown you write without generating commits. I realize that's probably not the answer you were hoping for. Unless you are writing very complex Markdown and HTML, I would think that previewing in VSCode or another local Markdown editor would catch any formatting issues before your commits.

**Reply by ahmedqfarrukh** [reply]
Got it. Will look into vscode plugins. Thank you again!

**Reply by mnel11** [reply]
Oh, thanks u!
